spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraJoinRDD.scala

```diff
@@ -34,7 +34,9 @@ class CassandraJoinRDD[L, R] private[connector](
     val where: CqlWhereClause = CqlWhereClause.empty,
     val limit: Option[Long] = None,
     val clusteringOrder: Option[ClusteringOrder] = None,
-    val readConf: ReadConf = ReadConf())(
+    val readConf: ReadConf = ReadConf(),
+    manualRowReader: Option[RowReader[R]] = None,
+    manualRowWriter: Option[RowWriter[L]] = None)(
```

Comment:
On the usages: Only the Java constructor requires us to use the param because it specifies the later parameters and doesn't understand named parameters. The Scala code can remain the same since all the later parameters are passed by name. 

(On removing the RWF from this class completely)
I tried implementing the above approaches originally and all of them ran into some unfortunate issues. I think the real way to go in the future (and this will require a much larger refactoring) would be to remove all of the code from the RDD itself except for the compute method and partitioning methods. This would leave the join RDD taking parameters of a CassandraConnector, a CQL Statement, and a RowReader and RowWriter. All of the api would be exposed through a secondary class which could correctly build the row readers and writers based on the options passed to the  factory. We have to do this because the row writer / reader can change dependent on the various options we allow in the api.

So my proposal for a future api would be

CassandraJoinRDDFactory ( leftSide: RDD , where, on, .... clauses, Implicit RWF, RRF ) WITH CassandraRDDApi 
CassandraJoinRDD( leftSide: Rdd, selectStatement: String, rowWriter: RowWriter, rowReader: RowReader, Implicit CassandraConnector,)

Then we would have an implicit  CassandraJoinRDDFactory -> CassandraJoinRDD 
 which would instantiate a JoinRDD from the factories

For this we would have to be careful about naming as not to disturb an user code.

I hesitate from fully doing this because that would be a much larger change, but if you would prefer that I could do that as well.(I do think it is the right long term situation)